2003-08-11  James E Wilson  <wilson@tuliptree.org>

	PR optimization/11319
	PR target/10021
	* alias.c (find_base_value, case REG): Return 0 not src if no base
	found.

--- gcc-3.3.1/gcc/alias.c.pr11319	2003-09-01 11:23:55.000000000 +0200
+++ gcc-3.3.1/gcc/alias.c	2003-09-01 11:34:37.000000000 +0200
@@ -797,7 +797,7 @@ find_base_value (src)
 	    return reg_base_value[regno];
 	}
 
-      return src;
+      return 0;
 
     case MEM:
       /* Check for an argument passed in memory.  Only record in the
